upf/upf_proxy_input.c

```diff
@@ -133,26 +134,48 @@ splice_tcp_connection (flow_entry_t * flow, flow_direction_t direction)
   session_t *s;
 
   if (rev->conn_index == ~0)
-    return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    {
+      vlib_increment_simple_counter (&gtm->upf_simple_counters
+				     [UPF_FLOW_NOT_STITCHED_NO_REV_CONNECTION],
+				     vlib_get_thread_index (), 0, 1);
+      return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    }
 
   if (flow->dont_splice)
-    return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    {
+      vlib_increment_simple_counter (&gtm->upf_simple_counters
+				     [UPF_FLOW_NOT_STITCHED_DONT_STITCH],
+				     vlib_get_thread_index (), 0, 1);
+      return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    }
 
   // lookup connections
   tc =
     transport_get_connection (TRANSPORT_PROTO_TCP, ftc->conn_index,
 			      ftc->thread_index);
   if (!tc)
-    return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    {
+      vlib_increment_simple_counter (&gtm->upf_simple_counters
+				     [UPF_FLOW_NOT_STITCHED_NO_TRANSPORT],
+				     vlib_get_thread_index (), 0, 1);
+      return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    }
 
   s = session_get_if_valid (tc->s_index, tc->thread_index);
   if (!s)
-    return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    {
+      vlib_increment_simple_counter (&gtm->upf_simple_counters
+				     [UPF_FLOW_NOT_STITCHED_INVALID_SESSION],
+				     vlib_get_thread_index (), 0, 1);
+      return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    }
 
   // check fifo, proxy Tx/Rx are connected...
   if (svm_fifo_max_dequeue (s->rx_fifo) != 0 ||
       svm_fifo_max_dequeue (s->tx_fifo) != 0)
-    return UPF_PROXY_INPUT_NEXT_TCP_INPUT;
+    vlib_increment_simple_counter (&gtm->upf_simple_counters
+				   [UPF_FLOW_NOT_STITCHED_FIFOS],
```

Comment:
`UPF_FLOW_NOT_STITCHED_FIFOS` is a misnomer here b/c we're ignoring the data in FIFOs and stitching the flows anyway. `UPG_FLOWS_STITCHED_DIRTY_FIFOS` perhaps?